### DISCUSSION TITLE
Help needed: how to access includes after calling to_models on Spaceship ConnectAPI response

### DISCUSSION THREAD

**Discussion by gsavit** [fastlane/fastlane#17633] [Q&A] [answered]
Hi all! I'm hoping someone can help out with this. I'm not so familiar with Ruby so this could definitely be user error, but I've been debugging for a while and can't figure this out.

Consider the following code:

```
response = Spaceship::ConnectAPI.get_app_store_versions(app_id: {appId}, includes: {includes}).all_pages
versions = response.flat_map(&:to_models)
```

Although I see the included objects in in the raw response's `included` array, when I flat map the response to models using the `to_models` method, all I end up with is `AppStoreVersion` objects, with none of the included objects. Where do the includes end up and how can I access them?

Thanks so much in advance!

**Comment by gsavit** [comment, marked as answer]
I just figured this out! It seems the `AppStoreVersion` class is missing attributes corresponding to the includes I'm trying to add. So the `update_attributes` call, which `to_models` uses, fails when trying to add the included objects. Is there some recommended way to address this?

**Reply by rogerluan** [reply]
@gsavit Hey 👋 

Sounds like what you need is to add the missing attributes to `AppStoreVersion` model, so they can be correctly mapped. PRs are always welcome ❤️ you can check how to get started [here](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md#i-want-to-contribute-to-fastlane) 😊

**Reply by gsavit** [reply]
Done! https://github.com/fastlane/fastlane/pull/17761
